server/controllers/dog.js

```diff
@@ -0,0 +1,128 @@
+const User = require("../models/User");
+const Dog = require("../models/Dog");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// @route GET /dogs/:id:
+// @desc Get dog for logged in user by :id
+// @access Private
+exports.loadDog = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
+  const dog = await Dog.findById(id);
+
+  if (dog) {
+    res.status(200).send({
+      success: {
+        dog,
+      },
+    });
+  } else res.status(400).json({ errors: "Bad request !" });
```

Comment:
If a dog hasn't been found, won't the status code here be a 404? :)

Reply:
I thought I would return **`404`** in case of resource does not exist(means the **`URL`** path is wrong) and I will return **`400`** only if the rest call is made with some invalid data **`(@Param || @body)`** for example

Reply:
Corrected successfully